### PR TITLE
Add missing gameplay constants

### PR DESCRIPTION
**Pull request by tobiasvl**
Convert remaining gameplay values to use equates

### CODE REVIEW

File: src/constants/gameplay.asm [hunk 1]
```diff
@@ -13,7 +13,8 @@ GAMEPLAY_FILE_DELETE       equ $04
 GAMEPLAY_FILE_COPY         equ $05
 GAMEPLAY_FILE_SAVE         equ $06
 GAMEPLAY_WORLD_MAP         equ $07
-GAMEPLAY_PEACH_PIC         equ $08 ; also used for the Eagle's Tower collapse cutscene
+GAMEPLAY_TOWER_COLLAPSE    equ $08
```
Comment (tobiasvl) [inline src/constants/gameplay.asm:16]: I think it makes sense to make this a different constant, although I'm open for discussion on that. I noticed that it seems like `$08` is also used as the value for the scene when Sale eats the canned food, although I'm not 100% sure of that yet.

File: src/constants/memory/wram.asm [hunk 2]
```diff
@@ -3353,7 +3353,7 @@ wSubtractHealthBuffer::
 wGameplayType::
   ds 1 ; DB95
 
-; Value depens on GameplayType: this can be a sequence index, or a frame counter.
+; Value depends on GameplayType: this can be a sequence index, or a frame counter.
```
Comment (kemenaran) [inline src/constants/memory/wram.asm:3356]: 🙇

File: src/constants/gameplay.asm (hunk 1 above)
Comment (kemenaran) [reply src/constants/gameplay.asm:16]: Hmm, having two constants is easier to read in the code – but I feel it also hides the complexity, by sort of pretending there are two different values, where there sharing the same.

If we were to refactor things, I think I'd slightly prefer a more generic name (`GAMEPLAY_SCENES`? `GAMEPLAY_CUTSCENES`? `GAMEPLAY_DEMO`?) over a duplicate constant definition. (Or leaving thinks as this, and embrace the terrifying hackiness of some things in the code base :p )

**Review by kemenaran — APPROVED** [review]
Nice, that must have been some work to look over all usages in the codebase.

File: src/constants/gameplay.asm (hunk 1 above)
Comment (tobiasvl) [reply src/constants/gameplay.asm:16]: Yeah, the thought did cross my mind when I found out the Sale scene probably uses the same value. `GAMEPLAY_CINEMATIC`?

Comment (kemenaran) [reply src/constants/gameplay.asm:16]: Yes, the few cinematics (Monkey's bridge, Banana eating, credits…) don't really use this.

This is more like a single-shot cutscene, or, hmm… full-screen-cutscene? Diorama?

**Review by kemenaran — APPROVED** [review]

**Pull request merged**